File: src/lambda/dms-switch/utils/list-stack-resources.ts
import { CloudFormationClient, ListStackResourcesCommand, StackResourceSummary } from '@aws-sdk/client-cloudformation';

export const listStackResources = async ({
  cf,
  resources = [],
  NextToken,
  StackName,
}: {
  cf: CloudFormationClient;
  StackName: string;
  resources: StackResourceSummary[];
  NextToken?: string;
}): Promise<StackResourceSummary[]> => {
  const listResourcesCommand = new ListStackResourcesCommand({
    StackName,
    NextToken,
  });
  const resourceSummaries = await cf.send(listResourcesCommand);
  const nextResources = [...resources, ...(resourceSummaries.StackResourceSummaries || [])];
  if (resourceSummaries.NextToken) {
    return listStackResources({ cf, StackName, resources: nextResources, NextToken: resourceSummaries.NextToken });
  } else {
    return nextResources;
  }
};

File: src/lambda/dms-switch-py/utils/list_stack_resources.py

def list_stack_resources(cf, stack_name, resources=None, next_token=None):
    if resources is None:
        resources = []
    response = cf.list_stack_resources(
        StackName=stack_name, NextToken=next_token)
    resource_summaries = response.get('StackResourceSummaries', [])
    resources += resource_summaries
    if 'NextToken' in response:
        return list_stack_resources(cf, stack_name, resources, response['NextToken'])
    else:
        return resources
